tests/rptest/tests/compatibility/arroyo_test.py

```diff
@@ -54,3 +54,5 @@ def test_arroyo_test_suite(self):
                 assert False, "Internal error during execution of Arroyo test suite"
             elif err.exit_status == 4:
                 assert False, "Pytest command line invocation error"
+        except Exception as err:
```

Comment:
It's not obvious why transforming the exception type to `AssertionError` helps: if there is an issue with teardown I think we'd still have it irrespective of the exception type we raise

Reply:
Ah this is a good point

Reply:
I initially thought that the exception is not caught by ducktape and teardown is skipped. That's not the case though. Since this is not a service it can evade the normal teardown process. I think John's suggestion of a finally block would work.